backend/Functions/Edna.AssignmentLinks/Edna.AssignmentLinks/AssignmentLinksApi.cs

```diff
@@ -103,18 +125,81 @@ public AssignmentLinksApi(ILogger<AssignmentLinksApi> logger, IMapper mapper)
         public async Task<IActionResult> DeleteLink(
             [HttpTrigger(AuthorizationLevel.Anonymous, "delete", Route = "assignments/{assignmentId}/links/{linkId}")] HttpRequest req,
             [Table(AssignmentLinksTableName)] CloudTable assignmentLinksTable,
-            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete)
+            [Table(AssignmentLinksTableName, "{assignmentId}", "{linkId}")] AssignmentLinkEntity entityToDelete,
+            [Assignment(AssignmentId = "{assignmentId}")] Assignment assignment,
+            [Platform] PlatformsClient platformsClient,
+            [LtiAdvantage] INrpsClient nrpsClient,
+            [Lti1] Lti1MembershipClient membershipClient)
         {
             if (entityToDelete == null)
+            {
+                _logger.LogInformation("Entity to delete is null");
                 return new OkResult();
+            }
+
+            IActionResult res = await ValidateUser(req, assignment, platformsClient, nrpsClient, membershipClient);
+            if (res.GetType() != typeof(OkResult))
+                return res;
 
             TableOperation deleteTable = TableOperation.Delete(entityToDelete);
             TableResult deleteResult = await assignmentLinksTable.ExecuteAsync(deleteTable);
 
             if (deleteResult.HttpStatusCode < 200 || deleteResult.HttpStatusCode >= 300)
                 return new InternalServerErrorResult();
 
+            _logger.LogInformation("Link deleted.");

```

Comment:
Same recommendation for this log entry as above. Getting an `http200` from this function might indicate this automatically.